app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -64,9 +74,40 @@ class MessagesController @Inject() (
     with ConvertError
     with UpscanResponseParser
     with ObjectStoreURIExtractor
+    with ContentTypeRouting
     with Logging {
 
-  def outgoing(eori: EoriNumber, movementType: MovementType, movementId: MovementId, messageId: MessageId): Action[Source[ByteString, _]] =
+  def outgoing(eori: EoriNumber, movementType: MovementType, movementId: MovementId, messageId: MessageId) =
+    contentTypeRoute {
+      case Some(_) => outgoingSmallMessage(eori, movementType, movementId, messageId)
+      case None    => outgoingLargeMessage(eori, movementType, movementId, messageId)
+    }
+
+  def outgoingLargeMessage(eori: EoriNumber, movementType: MovementType, movementId: MovementId, messageId: MessageId) =
+    Action.async {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+        (for {
+          messageType                 <- messageTypeExtractor.extractFromHeaders(request.headers).asPresentation
+          requestMessageType          <- filterRequestMessageType(messageType)
+          objectStoreResourceLocation <- extractObjectStoreURIHeader(request.headers)
+          objectSummary <- objectStoreService
+            .addMessage(
+              DownloadUrl(s"${appConfig.objectStoreUrl}/${request.headers.get(RouterHeaderNames.OBJECT_STORE_URI).get}"),
```

Comment:
`request.headers.get(RouterHeaderNames.OBJECT_STORE_URI).get` -- isn't this just `objectStoreResourceLocation`?

Reply:
request.headers.get(RouterHeaderNames.OBJECT_STORE_URI).get this will include ``` common-transit-convention-traders ``` and that is required as part of download url. but objectStoreResourceLocation does not contain ``` common-transit-convention-traders ``` as we are stripping `` common-transit-convention-traders ``` in extractor.

Reply:
So, should `objectStoreResourceLocation` contain two fields, one with the context path and one without, rather than having to pull down the header twice?

Reply:
yeah. we can do that way. No problem.